synapse/models/infotech.py

```diff
@@ -9,10 +13,50 @@
 import synapse.lib.chop as s_chop
 import synapse.lib.types as s_types
 import synapse.lib.module as s_module
+import synapse.lib.scrape as s_scrape
 import synapse.lib.version as s_version
 
 logger = logging.getLogger(__name__)
 
+# The CPE2.2 regex is not used in scrape but is used elsewhere. This seems like
+# the logical location for it since the CPE2.3 regex is here too.
+ALPHA = '[A-Za-z]'
+DIGIT = '[0-9]'
+UNRESERVED = r'[A-Za-z0-9\-\.\_]'
+SPEC1 = '%01'
+SPEC2 = '%02'
+# This is defined in the ABNF but not actually referenced
+# SPECIAL = f'(?:{SPEC1}|{SPEC2})'
+SPEC_CHRS = f'(?:{SPEC1}+|{SPEC2})'
+PCT_ENCODED = '%(?:21|22|23|24|25|26|27|28|28|29|2a|2b|2c|2f|3a|3b|3d|3e|3f|40|5b|5c|5d|5e|60|7b|7c|7d|7e)'
```

Comment:
I think this should be any hex value

Reply:
Per scrum discussion, I'm leaving this the same since expanding it would mean potentially allowing characters through that won't convert to CPE2.3. Since we do that conversion for every CPE2.2 string that comes in, allowing any percent encoded value would mean the norm pass would fail and the node won't be created anyway.